File: python-brain/schema.py
from pydantic import BaseModel, Field
from typing import List, Dict, Any, Optional

class EntitySchema(BaseModel):
    id: str
    type: str
    attributes: Dict[str, Any] = Field(default_factory=dict)
    
class RelationSchema(BaseModel):
    source: str
    target: str
    type: str
    attributes: Dict[str, Any] = Field(default_factory=dict)

class SimulationSchema(BaseModel):
    domain: str
    entities: List[EntitySchema]
    relations: List[RelationSchema]
    global_metrics: Dict[str, float] = Field(default_factory=dict)

File: rust-engine/src/schema.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EntitySchema {
    pub id: String,
    pub r#type: String,
    pub attributes: HashMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RelationSchema {
    pub source: String,
    pub target: String,
    pub r#type: String,
    pub attributes: HashMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SimulationSchema {
    pub domain: String,
    pub entities: Vec<EntitySchema>,
    pub relations: Vec<RelationSchema>,
    pub global_metrics: HashMap<String, f64>,
}
